substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/elf/dwarf/DwarfInfoSectionImpl.java

```diff
@@ -682,7 +682,8 @@ private int writeMethodDeclaration(DebugContext context, ClassEntry classEntry,
         String name = uniqueDebugString(range.getMethodName());
         log(context, "  [0x%08x]     name 0x%x (%s)", pos, debugStringIndex(name), name);
         pos = writeAttrStrp(name, buffer, pos);
-        int fileIdx = classEntry.localFilesIdx();
+        FileEntry fileEntry = range.getFileEntry();
+        int fileIdx = (fileEntry != null ? classEntry.localFilesIdx(fileEntry) : classEntry.localFilesIdx());
         log(context, "  [0x%08x]     file 0x%x (%s)", pos, fileIdx, range.getFileEntry().getFullName());
```

Comment:
`range.getFileEntry()` may be null here.

Reply:
Hi Foivos.
This is currently going through the gate so it would be better to fix this in a follow-up patch.
Perhaps you can fold this into your patch for #3419

Reply:
ACK, it's already fixed there (that's how I noticed it in the first place)!